source/GameData.cpp

```diff
@@ -193,6 +195,40 @@ void GameData::FinishLoading()
 
 
 
+void GameData::Clear()
+{
+	objects.Clear();
```

Comment:
You need to do a `background.Clear()` here as well. You will have to write a `Starfield::Clear()` function to go with it. 

The `Starfield` class will allocate some body objects that refer to sprites that you are about to delete. It will continue trying to draw them if the loading process takes too long.

Reply:
The new commit seems to have fixed this. Thanks!

Reply:
cool